x/gov/migrations/v046/store.go

```diff
@@ -40,12 +41,69 @@ func migrateProposals(store sdk.KVStore, cdc codec.BinaryCodec) error {
 	return nil
 }
 
-// MigrateStore performs in-place store migrations from v0.43 to v0.46. The
+// migrateVotes migrates all v1beta1 weighted votes (with sdk.Dec as weight)
+// to v1 weighted votes (with string as weight)
+func migrateVotes(store sdk.KVStore, cdc codec.BinaryCodec) error {
+	votesStore := prefix.NewStore(store, v042.VotesKeyPrefix)
+
+	iter := votesStore.Iterator(nil, nil)
+	defer iter.Close()
+
+	for ; iter.Valid(); iter.Next() {
+		var oldVote govv1beta1.Vote
+		err := cdc.Unmarshal(iter.Value(), &oldVote)
+		if err != nil {
+			return err
+		}
+
+		newVote := govv1.Vote{
+			ProposalId: oldVote.ProposalId,
+			Voter:      oldVote.Voter,
+		}
+		newOptions := make([]*govv1.WeightedVoteOption, len(oldVote.Options))
+		for i, o := range oldVote.Options {
+			newOptions[i] = &govv1.WeightedVoteOption{
+				Option: govv1.VoteOption(o.Option),
+				Weight: o.Weight.String(), // Convert to decimal string
+			}
+		}
+		newVote.Options = newOptions
+		bz, err := cdc.Marshal(&newVote)
+		if err != nil {
+			return err
+		}
+
+		// Set new value on store.
+		votesStore.Set(iter.Key(), bz)
+	}
+
+	return nil
+}
+
+// MigrateStore performs in-place store migrations from v2 (v0.43) to v3 (v0.46). The
 // migration includes:
 //
 // - Migrate proposals to be Msg-based.
 func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec) error {
 	store := ctx.KVStore(storeKey)
 
+	if err := migrateVotes(store, cdc); err != nil {
+		return err
+	}
+
 	return migrateProposals(store, cdc)
 }
+
+// Migrate_V046_4_To_V046_5 is a helper function to migrate chains from <=v0.46.6
+// to v0.46.7 ONLY.
+//
+// IMPORTANT: Please do not use this function if you are upgrading to v0.46
+// from <=v0.45.
+//
+// This function migrates the store in-place by fixing the gov votes weight to
+// be stored as decimals strings (instead of the sdk.Dec BigInt representation).
+//
+// The store is expected to be the gov store, and not any prefixed substore.
+func Migrate_V046_6_To_V046_7(store sdk.KVStore, cdc codec.BinaryCodec) error {
```

Comment:
I'm certain we don't want to run this.

Votes are deleted at the end of a voting period. This would only fix votes that happened prior to upgrade, but are for a proposal that people are still voting on. None of our networks have that.

The migration also looks like it doesn't properly account for votes cast after the upgrade. So running it has the chance of messing up such votes.

Any votes that might exist when we upgrade to v1.14 will have been written using the new gov stuff and have the correct format and not need migrating.

Reply:
> The migration also looks like it doesn't properly account for votes cast after the upgrade. So running it has the chance of messing up such votes.

Hey, chiming in here, what do you mean?